main/FatesRestartInterfaceMod.F90

```diff
@@ -2685,12 +2685,21 @@ subroutine get_restart_vectors(this, nc, nsites, sites)
              sites(s)%recruitment_rate(i_pft) = rio_recrate_sift(io_idx_co_1st+i_pft-1)
           enddo
 
-         !variables for fixed biogeography mode. These are currently used in restart even when this is off.
+          ! variables for fixed biogeography mode. These are currently used in restart even when this is off.
           do i_pft = 1,numpft
              sites(s)%use_this_pft(i_pft) = rio_use_this_pft_sift(io_idx_co_1st+i_pft-1)
              sites(s)%area_pft(i_pft)     = rio_area_pft_sift(io_idx_co_1st+i_pft-1)
           enddo
 
+          ! calculate the bareground area for the pft in no competition modes
+          if (hlm_use_nocomp .eq. itrue) then
+             if (sum(sites(s)%area_pft(1:numpft)) .lt. area) then
```

Comment:
That's a good idea.  I was lazy basically copy/pasted this from EDInitMod.  Looking back at the `set_site_properties` procedure, I think we probably should add a check to make sure the sum is not larger than `area`.

Reply:
If I add a check to make sure the sum is never greater that `area` during prior to the `get_restart_vectors`, do you think that's sufficient since we don't update `area_pft` after the initialization of the run?

Reply:
After reviewing the checks for the variables that set `area_pft` .  We check that [`hlm_pft_map` sums to unity in EDPftvarcon](https://github.com/NGEET/fates/blob/e216ca0006abd371d55f6b0643f09dfeabd1d7c3/main/EDPftvarcon.F90#L1761) and that [`pft_areafrac` does the same in clmfates_interfacemod](https://github.com/ESCOMP/CTSM/blob/c6d8032c9224e90978c8f73b8ddde23b41671671/src/utils/clmfates_interfaceMod.F90#L715).  So we shouldn't even be able to get a value for `area_pft` less than zero after multiplying by `area`